src/_pytest/assertion/rewrite.py

```diff
@@ -52,6 +53,8 @@
 PYC_EXT = ".py" + (__debug__ and "c" or "o")
 PYC_TAIL = "." + PYTEST_TAG + PYC_EXT
 
+_SCOPE_END_MARKER = object()
```

Comment:
Minor, but how about we make a specific subclass from `ast.AST`, so the typing for `scope: tuple[ast.AST, ...] = ()` is consistent?

```suggestion
class ScopeEnd(ast.AST): 
    """
    (some docs here).
    """
```

And then instead of checking ` if node == _SCOPE_END_MARKER:`. we can `if isinstance(node, ScopeEnd)`.

Reply:
The `self.scope` typing is correct with `tuple[ast.AST, ...]` already. `_SCOE_END_MARKER` is only added to the `nodes` list never to `self.scope`.